wordpress-mobile/WordPress-Android
# Issue/11906 reader move tab layout to parent part 6.5 tracking

Fixes #11906
Partially fixes #11849

This is another PR in a series of PRs which is trying to introduce a regular TabLayout+ViewPager2 in the main Reader view. The current implementation uses FilteredRecyclerView component that has some downsides (eg. swipe gesture doesn't work). These changes will also help us implement the improvements in the ReaderImprovements project.

P.S. The known issues/remaining tasks are listed in the ReaderImprovements project board.

Goal of this PR is to fix ReaderTracker behavior which was broken during the refactoring.

The application_closed event should contain `time_in_reader_filtered_list`, `time_in_reader_paged_post`, `time_in_main_reader` properties with number of ms spent in the corresponding screens of the Reader, where:

time_in_main_reader : Number of seconds spent on the main Reader view (the one with "following/discover/liked/saved" tabs + search. 

time_in_reader_filtered_list : Number of seconds spend on the "Following tab" while the subfilter is in use - in other words when the subfilter contains a value.

time_in_reader_paged_post : Number of seconds in the post detail - to be more specific ->a  viewpager of post details.


To test:
**Prerequisites - enable logging in the App Settings -> Privacy Settings -> Collect information.**

- Walk through different sections of Reader and observe `ReaderTracker: started/stopped` events in the LogCat. 
- Observe stopped event is/are invoked when you switch selected tab in the BottomNavigationBar or when you leave the app
- When you leave the app - by pressing home screen or by pressing "back", notice the `application_closed` with corresponding properties and recored time is logged in LogCat 

PR submission checklist:
